lidar:
  - driver:
      pcap_play_synchronization: true # pcap play rate synchronize with the host time
      udp_port: 2368 # UDP port of lidar
      ptc_port: 9347 # PTC port of lidar
      device_ip_address: 192.168.1.201 # host_ip_address
      pcap_path: "Your pcap file path" # The path of pcap file
      correction_file_path: "$PROJECT_PATH/src/driver/HesaiLidar_SDK_2.0/correction/angle_correction/PandarXT_Angle Correction File.csv" #The path of correction file
      firetimes_path: "$PROJECT_PATH/src/driver/HesaiLidar_SDK_2.0/correction/firetime_correction/PandarXT_Firetime Correction File.csv" #The path of firetimes file
      enable_distance_correction: true # Enable distance correction to account for offset between sensor origin and laser emitter/receiver
      source_type: 1 #The type of data source, 1: real-time lidar connection, 2: pcap, 3: packet rosbag
      frame_start_azimuth:
        -1 #Frame azimuth for Pandar128, range from 1 to 359, set it less than 0 if you
        #do not want to use it.
      #transform param
      x: 0
      y: 0
      z: 0
      roll: 0
      pitch: 0
      yaw: 0
    ros:
      ros_frame_id: hesai_lidar #Frame id of packet message and point cloud message
      ros_recv_packet_topic: /lidar_packets #Topic used to receive lidar packets from rosbag
      ros_send_packet_topic: /lidar_packets #Topic used to send lidar raw packets through ROS
      ros_send_point_cloud_topic: /points #Topic used to send point cloud through ROS
      send_packet_ros: false #true: Send packets through ROS
      send_point_cloud_ros: true #true: Send point cloud through ROS
